src/renderer/commands/quickOpen.js

```diff
@@ -0,0 +1,196 @@
+import path from 'path'
+import { ipcRenderer } from 'electron'
+import { isChildOfDirectory } from 'common/filesystem/paths'
+import bus from '../bus'
+import { delay } from '@/util'
+import FileSearcher from '@/node/fileSearcher'
+
+const MD_EXTENSION = /\.markdown|\.mdown|\.mkdn|\.md|\.mkd|\.mdwn|\.mdtxt|\.mdtext|\.text|\.txt$/i
+const SPECIAL_CHARS = /[\[\]\\^$.\|\?\*\+\(\)\/]{1}/g // eslint-disable-line no-useless-escape
+
+// The quick open command
+class QuickOpenCommand {
+  constructor (rootState) {
+    this.id = 'file.quick-open'
+    this.description = 'File: Quick Open'
+    this.shortcut = null
+
+    this.subcommands = []
+    this.subcommandSelectedIndex = -1
+
+    // Reference to folder and editor and project state.
+    this._editorState = rootState.editor
+    this._folderState = rootState.project
+
+    this._directorySearcher = new FileSearcher()
+    this._cancelFn = null
+  }
+
+  search = async query => {
+    // Show opened files when no query given.
+    if (!query) {
+      return this.subcommands
+    }
+
+    const { _cancelFn } = this
+    if (_cancelFn) {
+      _cancelFn()
+      this._cancelFn = null
+    }
+
+    const timeout = delay(300)
+    this._cancelFn = () => {
+      timeout.cancel()
+      this._cancelFn = null
+    }
+
+    await timeout
+    return await this._doSearch(query)
+  }
+
+  run = async () => {
+    const { _editorState, _folderState } = this
+    if (!_folderState.projectTree && _editorState.tabs.length === 0) {
+      throw new Error(null)
+    }
+
+    this.subcommands = _editorState.tabs
+      .map(t => t.pathname)
+      // Filter untitled tabs
+      .filter(t => !!t)
+      .map(pathname => {
+        const item = { id: pathname }
+        Object.assign(item, this._getPath(pathname))
+        return item
+      })
+  }
+
+  execute = async () => {
+    // Timeout to hide the command palette and then show again to prevent issues.
+    await delay(100)
+    bus.$emit('show-command-palette', this)
+  }
+
+  executeSubcommand = async id => {
+    const { windowId } = global.marktext.env
+    ipcRenderer.send('mt::open-file-by-window-id', windowId, id)
+  }
+
+  unload = () => {
+    this.subcommands = []
+  }
+
+  // --- private ------------------------------------------
+
+  _doSearch = async query => {
+    this._cancelFn = null
```

Comment:
You already set `this._cancelFu` to `null` in `search` method.

Reply:
That's not right because it's set `timeout` cancel function to `null` - see L42.